### ISSUE TITLE
DST misunderstanding - Mismatch between theory and implementation in WorstCaseEstimationLoss

### ISSUE THREAD

**Issue by emasa** [thuml/Transfer-Learning-Library#206]
Hi dear authors,

My concern is about the DST (Debiased Self-Training for Semi-Supervised Learning).

I found a difference between Eq  (8) in the main paper and the implementation of WorstCaseEstimationLoss.

In the paper, the last term of Eq  (8), L_l(ψ, h') (also defined in Eq (1)) is a supervised loss, which trains the worst-case classifier with **ground truth labels**. However, in WorstCaseEstimationLoss, the targets for this loss are obtained from logits estimated with the main head classifier (potentially noisy). The documentation reads:
`        - y_l: logits output `y^l` by the main head on labeled data`
      
Later on, this is used in:
```
        _, prediction_l = y_l.max(dim=1)
        loss_l = self.eta_prime * F.cross_entropy(y_l_adv, prediction_l)
```
This difference is very noticeable, as the head classifier might be imperfect, especially at the beginning of training. Why not use clean labels in the implementation? Is there a notation issue in the main paper, is it an implementation trick or is it a bug?  

Looking forward to clarifying this issue and nice job with the repo!

**Comment by thucbx99** [comment]
Hi, the implementation here is just to make the two losses (loss_l and loss_u) look more symmetrical.

As the head classifier is always trained with **_ground truth labels_**, _**prediction_l**_ will quickly approach **_labels_l_** during training. Indeed, in the early stage of training, the head classifier might be imperfect. And this might be harmful to learning. However, in our experiments, we found that using _**prediction_l**_ or **_labels_l_** works similarly well.
